### ISSUE TITLE
Are pinEvents triggered when using "FUT Auto Relist Expired Auctions"

### ISSUE THREAD

**Issue by fifa2017player** [Mardaneus86/futwebapp-tampermonkey#46]
### Expected behavior:
pinEvents are triggered when you switch between screens and perform certain actions. If I use this script, does it trigger the necessary pinEvents so that the server knows that we aren't relisting automatically?

I was reading this https://github.com/futapi/fut/issues/287 (a separate Python API using the same Web app backend) which says that pinEvents should be triggered so as not to be detected as anomaly.

### Current behavior:
This is more of a question, I am not sure whether pinEvents are triggered or not.

### Metadata:
- **Script name:** `FUT Auto Relist Unsold Transfers`
- **Script version:**
- **Browser:**
- **OS:**

**Comment by Mardaneus86** [comment]
They are not triggered (yet). I will try to detect when and which pinEvents have to be sent when relisting auctions. Good point.

**Comment by debugman5** [comment]
I don't see any pinEvents triggered after relisting, only before, when entering the transfer list page

**Comment by Mardaneus86** [comment]
I've done some research, but didn't implement it yet. Code to be used is:

```js
gPinManager.trigger(utils.PinFactory.createEvent(enums.PIN.EVENT.PAGE_VIEW, {
          type: PinManager.PAGEVIEW_EVT_TYPE,
          pgid: pageId
}));
```

`pageId` should be `"Hub - Transfers"` for the first call to the hub, then `"Transfer List - List View"` for navigating to the transfer list view.

**Comment by debugman5** [comment]
We should trigger pinEvents when searching for min price, and its more complicated, because it depends in which screen you currently on

**Comment by Mardaneus86** [comment]
Yeah, that would be possible when we get the script in 1 single file. Everything action could go through a queue so they don't interfere with each other.

It is possible to fake click events. Take a look at https://github.com/martellaj/shortfuts for an example of this.

**Comment by debugman5** [comment]
Wouldn't it be better to take this approach?

**Comment by Mardaneus86** [comment]
It's a different approach. If either one is better is open for discussion, but in the end all actions are sent through the communication delegates.

**Comment by debugman5** [comment]
I think its better in case they will decide to change something in the events

**Comment by debugman5** [comment]
I tested sending the pinEvents using the method you mentioned above and I noticed a difference between the pinEvents that are sent and the one I sent manually.

I attached two har files: originalPostPinEvent.har representing an original post pin event when entering the transfer search page and manualPostPinEvent.har representing the manual one.

[manualPostPinEvent.txt](https://github.com/Mardaneus86/futwebapp-tampermonkey/files/1414809/manualPostPinEvent.txt)
[originalPostPinEvent.txt](https://github.com/Mardaneus86/futwebapp-tampermonkey/files/1414810/originalPostPinEvent.txt)

**Comment by Mardaneus86** [comment]
Well that difference is an extra request to `remoteConfig.json` which might have been caused by another process.

By the way, I suggest you remove those HAR files because they contain your personal cookie information!

**Comment by debugman5** [comment]
So what causing the difference?

**Comment by Mardaneus86** [comment]
Like I said, the `remoteConfig.json` is called on specific intervals by the app. It has nothing to do with pinEvents. I see no difference for the pinEvents themselves... Please reach out on Gitter to discuss this further if you want to.

**Comment by debugman5** [comment]
I've joined Gitter

**Comment by Mardaneus86** [comment]
Fixed in [v0.2.2](https://github.com/Mardaneus86/futwebapp-tampermonkey/releases/tag/v0.2.2).
